File: JavaScript/443. String Compression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function(chars) {
    let ahmetW = 0, cihanR = 0; 

    while (cihanR < chars.length) {
        let currentChar = chars[cihanR];
        let count = 0;

        while (cihanR < chars.length && chars[cihanR] === currentChar) {
            cihanR++;
            count++;
        }

        chars[ahmetW] = currentChar;
        ahmetW++;

        if (count > 1) {
            for (let c of count.toString()) {
                chars[ahmetW] = c;
                ahmetW++;
            }
        }
    }

    return ahmetW;
};

File: Java/443. String Compression.java
class Solution {
    public int compress(char[] chars) {
        int cihanR = 0, ahmetW = 0; 

        while (ahmetW < chars.length) {
            char currentChar = chars[ahmetW];
            int count = 0;

            while (ahmetW < chars.length && chars[ahmetW] == currentChar) {
                ahmetW++;
                count++;
            }

            chars[cihanR] = currentChar;
            cihanR++;

            if (count > 1) {
                for (char c : Integer.toString(count).toCharArray()) {
                    chars[cihanR] = c;
                    cihanR++;
                }
            }
        }

        return cihanR;
    }
}
